src/lib.ts

```diff
@@ -23,18 +23,22 @@ export async function getInternalLinks(
 
   const analyzedLinks: AnalyzedLink[] = [];
 
-  // I won't worry about using Promise.all with some kind of batching here, maybe refactor later.
-  for (const nodePath of nodePaths) {
-    // eslint-disable-next-line no-await-in-loop
-    const contents = await fs.readFile(nodePath, 'utf-8');
+  const promiseList = nodePaths.map((nodePath) => fs.readFile(nodePath, 'utf-8'));
+  const contentList: string[] = [];
+  
+  await Promise.all(promiseList)
```

Comment:
The `contentList` variable could easily be derived from `await`ing the `Promise.all` return value:

```
// todo: maybe batch?
const promiseList = nodePaths.map((nodePath) => fs.readFile(nodePath, 'utf-8'));
const contentList = await Promise.all(promiseList);
```